metadata-ingestion/src/datahub/utilities/memory_leak_detector.py

```diff
@@ -103,21 +100,12 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
 
         try:
             res = func(*args, **kwargs)
-        # Catch all exceptions, including SystemExit that some of the commands raise via sys.exit()
-        except BaseException as e:
-            exception = e
-
-        if detect_leaks:
-            _perform_leak_detection()
-            logger.info(
-                f"Finished memory leak detection on command: {func.__module__}.{func.__name__}"
-            )
-
-        if exception is None:
             return res
-        elif isinstance(exception, SystemExit):
-            sys.exit(exception.code)
-        else:
-            raise exception
+        finally:
+            if detect_leaks:
+                _perform_leak_detection()
+                logger.info(
+                    f"Finished memory leak detection on command: {func.__module__}.{func.__name__}"
+                )
```

Comment:
do we ever need to call `tracemalloc.stop()`?

Reply:
Yes, https://github.com/linkedin/datahub/pull/4363/files#diff-4187fa65143e12d74b8891b8eeb60f35a0e8be8c2488da9bbb2be2aab3d8ad8bR86

Reply:
ah thanks missed that